docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,93 @@
+# Satellite Service Separation
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+Currently the Satellite is a single binary made of up the following services:
+- version
+- overlay
+- metainfo
+- inspector
+- orders
+- repair (includes checker and repairer)
+- audit
+- garbage collection (GC)
+- accounting (includes tally and rollup service and the live accounting cache)
+- mail
+- console
+- nodestats
+- note: marketing service is currently on pause and is not included in this doc
+- note: kademlia, discovery, and vouchers are being removed and not included in this doc
+
+The Satellite has the following public GRPC endpoints:
+- metainfo
+- nodestats
+- orders
+
+The Satellite has the following privare GRPC endpoints:
+- inspectors
+
+All services (except version) make connections to the masterDB. Five services rely on metainfo service to access the metainfoDB, this includes inspectors, accounting, audit, repair, and garbage collection.
+
+The current items that prevent Satellite horizontal scaling include:
+- live accounting cache is currently stored in memory which cannot be shared across many replicas
+- in-memory databases (i.e. boltDB and sqlite)
+- potentially things that use mutexes, sync.Mutex or sync.RWMutex
+- set on database, i.e. non transactional db changes
+- ?
+
+## Design
+
+The plan here is to break the Satellite into many processes that can run independently in their own isolated environment (i.e. VM or container).  These isolated services should all be able to run replicas and load balance between them. So this means they need to share access to any persistent data.
+
+Currently there is only one Satellite binary. We propose to add the following binaries:
+
+1. api binary which includes the following: all public grpc endpoints, overlay, metainfo, nodestats
+2. private api binary which includes: all private grpc endpoints, inspectors
+3. console binary which includes: mail, console, overlay, metainfo
+4. repair binary which includes: overlay, metainfo, orders
+5. audit binary which includes: overlay, metainfo
+6. accounting binary which includes: tally and rollup, overlay, metainfo
+7. garbage collection binary which includes: overlay, metainfo
+8. uptime binary which inludes: overlay
+  - note: there is an ongoing discussion about the uptime service so this might change
+
+The following diagram shows the above propsed design:
+
+![Diagram of the above listed binaries](sa-separation-design.png)
+
+## Rationale
+
+WIP
+
+## Implementation
+
```

Comment:
Let's add a step before this to introduce `SatelliteSystem` in `testplanet`, such that it would be separate from actually making a particular service process.